src/handlers/LedgerWalletClient.ts

```diff
@@ -0,0 +1,170 @@
+export async function createLedgerAccount({ deviceId, index = 0 }: CreateLedgerAccountParams): Promise<Account> {
+  // Derivation path (m/44'/60'/0'/0/index)
+  const path = getHdPath({ type: WalletLibraryType.ledger, index });
+
+  // Lazily instantiate the Ledger ETH application
+  const ethPromise = getEthApp(deviceId);
+
+  // Resolve Ledger address up-front as the account `address` must be synchronous
+  const accountResponse = await retryLedgerCall(ethPromise, eth => eth.getAddress(path));
+  const address = getAddress(accountResponse.address) as Address;
+
+  /**
+   * Helper that returns a closure over the Ledger retry logic for each sign method
+   */
+  const withRetry = <T>(fn: (eth: AppEth) => Promise<T>) => retryLedgerCall(ethPromise, fn);
+
+  const account: Account = {
+    address,
+    async signMessage({ message }: { message: Bytes | string | Hex }) {
+      // Convert input to bytes then to a hex string without the 0x prefix – the format expected by Ledger
+      if (typeof message === 'string') {
+        message = toUtf8Bytes(message);
+      }
+      // `hexlify` will handle Uint8Array / BytesLike inputs.
+      const messageHex = hexlify(message as Bytes).substring(2);
+      const sig = await withRetry(eth => eth.signPersonalMessage(path, messageHex));
+      sig.r = '0x' + sig.r;
+      sig.s = '0x' + sig.s;
+      return joinSignature(sig) as Hex;
+    },
+    async signTypedData({ domain, types, primaryType, message, version = SignTypedDataVersion.V4 as any }: any) {
+      // Sanitize & hash typed-data exactly like ethers implementation
+      const domainSeparatorHex = TypedDataUtils.hashStruct('EIP712Domain', domain, types, version).toString('hex');
+      const hashStructMessageHex = TypedDataUtils.hashStruct(primaryType, message, types, version).toString('hex');
+      const sig = await withRetry(eth => eth.signEIP712HashedMessage(path, domainSeparatorHex, hashStructMessageHex));
+      sig.r = '0x' + sig.r;
+      sig.s = '0x' + sig.s;
+      return joinSignature(sig) as Hex;
+    },
+    async signTransaction(tx: TransactionSerializable) {
+      // Serialize the unsigned transaction to hex (without 0x prefix) for Ledger
+      const unsignedTx = viemSerializeTransaction(tx).slice(2);
+
+      // Provide Ledger with additional plugin resolution context
+      const resolution = await withRetry(eth =>
+        ledgerService.resolveTransaction(unsignedTx, eth.loadConfig, {
+          erc20: true,
+          externalPlugins: true,
+          nft: true,
+        })
+      );
+
+      // Sign the transaction on the Ledger device
+      const sig = await withRetry(eth => eth.signTransaction(path, unsignedTx, resolution));
+
+      // Combine the signature with the original transaction
+      const signedTx = viemSerializeTransaction(tx, {
+        r: `0x${sig.r}` as Hex,
+        s: `0x${sig.s}` as Hex,
+        v: BigInt(`0x${sig.v}`),
+      }) as Hex;
+
+      return signedTx;
+    },
+  } as unknown as Account;
```

Comment:
pretty dangerous, depending on what's the issue here it might be worth resolving